From 0e1d0f0cb9f68a0fc25bd429ce97849d5bd6d7b1 Mon Sep 17 00:00:00 2001
From: Sanyog Kale <skale@habana.ai>
Date: Tue, 18 Oct 2022 09:36:58 +0300
Subject: [PATCH] [SW-0] gaudi3: perform sku reconfiguation before slave fuse init

Currently, in fuse emulation flow, master die performs slave
fuse emulation init and then perform sku assignment based on fuse values
from emulation (ie. SPI flash). As sku check is done in slave fuse init
flow, this patch moves assignment of sku before slave fuse emulation
init.

Change-Id: Icc2f83f9dc70b24896c53dd28582ca461db395a1
---

diff --git a/arch/gaudi3/arc/common.c b/arch/gaudi3/arc/common.c
index 18175af..166c811 100644
--- a/arch/gaudi3/arc/common.c
+++ b/arch/gaudi3/arc/common.c
@@ -624,17 +624,6 @@
 
 	if (arcboot_rom_fake_fuse_en() || arcboot_rom_rma_mode_en()) {
 		spi_flash_fuse_init();
-		/*
-		 * Re-configure sku value based on new config_fuse,
-		 * Don't re-configure if original sku value is 1.
-		 */
-		if (arcboot_rom_global.sku)
-			return;
-
-		if (arcboot_rom_global.config_fuse & EFUSE_SYSCFG_SKU_MASK)
-			arcboot_rom_global.sku = 1;
-		else
-			arcboot_rom_global.sku = 0;
 	}
 }
 
diff --git a/arch/gaudi3/arc/spi_flash.c b/arch/gaudi3/arc/spi_flash.c
index 7526561..a471d98 100644
--- a/arch/gaudi3/arc/spi_flash.c
+++ b/arch/gaudi3/arc/spi_flash.c
@@ -108,6 +108,13 @@
 	fuse_read(EFUSE_SYSCFG_OFFSET, &arcboot_rom_global.config_fuse);
 	fuse_read(EFUSE_SYSCFG2_OFFSET, &arcboot_rom_global.config2_fuse);
 
+	/*
+	 * Re-configure sku value based on new config_fuse,
+	 * Don't re-configure if original sku value is 1.
+	 */
+	if ((!arcboot_rom_global.sku) && (arcboot_rom_global.config_fuse & EFUSE_SYSCFG_SKU_MASK))
+		arcboot_rom_global.sku = 1;
+
 	spi_flash_slave_init();
 
 	/* Since spi fuse init is successful, set fuse_emulation_done to true */
